src/biconnectivity/articulation.jl

```diff
@@ -69,12 +19,68 @@ julia> articulation(PathGraph(5))
  4
 ```
 """
-function articulation(g::AbstractGraph)
-    state = Articulations(g)
-    for u in vertices(g)
-        if state.depth[u] == 0
-            visit!(state, g, u, u)
+function articulation(g::SimpleGraph{T}) where T
```

Comment:
Can you change this so it is restricted to undirected abstract graphs? (using `@traitfn`)